### DISCUSSION TITLE
Is VSCode Explorer a component reusable in my own extension?

### DISCUSSION THREAD

**Discussion by marcoparenzan** [microsoft/vscode-discussions#1068] [Extension Development QnA] [answered]
Hello everyone!

I’m developing with a student a VSCode Extension that present a sidebar widget. It contains a list of names. I’d like to have on the same sidebar a file explorer which I can configure to browse a folder I can specify. I attach an image with the current status of the extension.

![VSCode with explorer](https://github.com/microsoft/vscode-discussions/assets/76774/79a6642f-c32d-480f-9aed-8faa72288b4a)

In practice, with the extension i need to browse selected specific folders...when I click on the specific entry of the extension.

I’m not able to find if it is possible to reuse/include the file browse component of VSCode in my extension. I suppose that file browser is a component that can be reused (perheaps no…. 😊).

Sorry if there is already a similar discussion, but I have just started vscode dev and just discovered this discussion forum (thanks Olivia! :))

Thanks in advance for any help.

**Comment by alefragnani** [comment, marked as answer]
Hi @marcoparenzan ,

You can’t reuse the Explorer view on your own extensions, unfortunately.

Instead, you may use the `Tree View API` (https://code.visualstudio.com/api/extension-guides/tree-view), which allows you to  replicate most of its behavior. There is a good sample in the VS Code Extensions Sample repo (https://github.com/microsoft/vscode-extension-samples/tree/main/tree-view-sample). 

And, if you feel the `Tree View API` to be not as powerful (on a UI/UX standpoint), you may look for the `Webview API` (https://code.visualstudio.com/api/extension-guides/webview). In this case will you have more freedom on how your Explorer looks/behave, but with a bit more complexity under the hood. 

Which one to use, depends on your needs.

Hope this helps
